drivers/SmartThings/zigbee-window-treatment/src/test/test_zigbee_window_treatment_aqara_curtain_driver_e1.lua

```diff
@@ -0,0 +1,368 @@
+    })
+    test.socket.zigbee:__expect_send({
+        mock_device.id,
+        PowerConfiguration.attributes.BatteryPercentageRemaining:configure_reporting(mock_device, 30, 3600, 1)
+    })
+    test.socket.zigbee:__expect_send({
+        mock_device.id,
+        zigbee_test_utils.build_bind_request(mock_device, zigbee_test_utils.mock_hub_eui, WindowCovering.ID)
+    })
+    test.socket.zigbee:__expect_send({
+        mock_device.id,
+        WindowCovering.attributes.CurrentPositionLiftPercentage:configure_reporting(mock_device, 0, 600, 1)
+    })
+    test.socket.zigbee:__expect_send({
+        mock_device.id,
+        Groups.server.commands.RemoveAllGroups(mock_device)
+    })
+    test.socket.zigbee:__expect_send({
+        mock_device.id,
+        zigbee_test_utils.build_attribute_read(mock_device, PRIVATE_CLUSTER_ID, { PRIVATE_CURTAIN_RANGE_FLAG_ATTRIBUTE_ID }, MFG_CODE)
+    })
+    test.socket.zigbee:__expect_send({
+        mock_device.id,
+        zigbee_test_utils.build_attribute_read(mock_device, PRIVATE_CLUSTER_ID, { PRIVATE_CURTAIN_LOCKING_STATUS_ATTRIBUTE_ID }, MFG_CODE)
+    })
+    test.socket.zigbee:__expect_send({
+        mock_device.id,
+        WindowCovering.attributes.CurrentPositionLiftPercentage:read(mock_device)
+    })
+    test.socket.zigbee:__expect_send({
+        mock_device.id,
+        PowerConfiguration.attributes.BatteryPercentageRemaining:read(mock_device)
+    })
+    mock_device:expect_metadata_update({ provisioning_state = "PROVISIONED" })
+  end
+)
+
+test.register_coroutine_test(
+  "Handle reverseCurtainDirection in infochanged",
+  function()
+    test.socket.device_lifecycle:__queue_receive(mock_device:generate_info_changed({
+      preferences = { ["stse.reverseCurtainDirection"] = true }
+    }))
+    test.socket.zigbee:__expect_send({
+      mock_device.id,
+      custom_write_attribute(mock_device , WindowCovering.ID, WindowCovering.attributes.Mode.ID,
+      data_types.Bitmap8, 0x01, nil)
+    })
+  end
+)
+
+test.register_coroutine_test(
+  "Handle softTouch in infochanged",
+  function()
+    test.socket.device_lifecycle:__queue_receive(mock_device:generate_info_changed({
+      preferences = { ["stse.softTouch"] = true }
+    }))
+    test.socket.zigbee:__expect_send({ mock_device.id,
+      cluster_base.write_manufacturer_specific_attribute(mock_device, PRIVATE_CLUSTER_ID,
+      PRIVATE_CURTAIN_MANUAL_ATTRIBUTE_ID, MFG_CODE, data_types.Boolean, false) })
+  end
+)
+
+
+test.register_coroutine_test(
+  "Window shade open cmd handler",
+  function()
+    test.socket.capability:__queue_receive(
+      {
+        mock_device.id,
+        { capability = "windowShade", component = "main", command = "open", args = {} }
+      }
+    )
+    -- test.socket.zigbee:__expect_send({
+    --     mock_device.id,
+    --     WindowCovering.server.commands.UpOrOpen(mock_device, 'open')
+    -- })
+  end
+)
```

Comment:
you need to write this test so that the initialization happens so that the test can verify the command is actually sent in that case

or, ideally, you have a test for both possibilities: initialized and uninitialized

Reply:
modified